### PR TITLE
Identify API errors and send error response

### LINKED ISSUE #2975: Saving an owned brew while not logged in causes perpetual spinning save icon
If you somehow reach the edit page of a brew without being logged in properly as one of the authors (you signed out while the tab was already open, you are removed as an author while the document is open, etc.), making an edit and then trying to save will give you a perpetual spinning save icon, instead of any warning that you need to be logged in.

This is not clear to the user that they need to refresh their login.

For reference, the server logs this error:

image

And the client browser simply gets this:

image

Coming from this line:
image

It shouldn't be possible for this to happen under most normal use, but it can lead to lost work if the user isn't getting a proper warning that the save is failing.

### PR DESCRIPTION
**Pull request by G-Ambatte**
This PR does not fully resolve #2975, #2510, but does unblock them.

This PR adds an identification method to the error handling middleware function and identifies errors arising from API calls (via `req.originalUrl.startsWith('/api/')`), which causes an error response to be sent (`res.status().send()`) rather than `res.send(page)`.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -472,9 +472,18 @@ const getPureError = (error)=>{
 };
 
 app.use(async (err, req, res, next)=>{
-	const status = err.status || err.code || 500;
+	err.originalUrl = req.originalUrl;
 	console.error(err);
 
+	if(err.originalUrl?.startsWith('/api/')) {
+		// console.log('API error');
+		res.status(err?.status || err?.response?.status || 500).send(err.message || err);
```
Comment (Gazook89) [inline server/app.js:480]: Do you need the `?` operator on `err?.status` and `err?.response`?  Presumably there *is* an error if you've entered this `if...then` statement.  

If anything, they could be replaced by throwing the `?` in the `if(err?.originalURL....` condition?

**Review by Gazook89 — APPROVED** [review]
The server side stuff is usually beyond me, though this looks a simple PR.  Looks good to me (though I left one comment).

File: server/app.js (hunk 1 above)
Comment (G-Ambatte) [reply server/app.js:480]: I think that entire line came directly from `homebrew.api.js`, you are correct - there is no way that we can enter that branch of the `if` with a non-existent `err`.

**Comment by calculuschild** [comment]
Ok, this looks good! It does unstick #2975 by allowing the failed request to actually return an error instead of nothing, but to fully resolve the issues mentioned above, they need to have a custom error message displayed in `error-navitem.jsx` instead of the generic "oops! problem saving".

Thanks @G-Ambatte! 💯 ⭐ 🪄

**Pull request merged**